app/models/concerns/authorization.rb

```diff
@@ -0,0 +1,17 @@
+# frozen_string_literal: true
+
+module Authorization
+  extend ActiveSupport::Concern
+
+  included do
+    before_action :verify_permissions
+  end
+
+  def verify_permissions
+    permission = Permission.find_by(resource: controller_name, action: action_name)
+    unless permission.in? current_user&.permissions
+      flash[:error] = "You don't have permission to perform this action on this resource."
+      redirect_to root_url, status: :forbidden
+    end
+  end
+end
```

Comment:
maybe unit test this?

Reply:
Could do, but it will always be tested in integration when doing controller tests, see `posts_controller_tests.rb`. I don't really feel like I need to have this tested by itself for now.

Reply:
So mocking boundaries on the controller level is impossible in rails?

Reply:
I don't think it's impossible, why do you ask? I said I could test this in isolation but I don't see the value as of now. Neither controllers nor this concern are too complex to have their own suite of unit tests, and both behaviors are well encapsulated and reflected in the controllers' integration tests. I think the plus value ratio of time invested/confidence gained of unit testing either controller or concerns right now is minimal compared to the bang-for-bucks we get out of integration tests.